src/main/java/org/apache/sling/i18n/impl/JcrResourceBundleProvider.java

```diff
@@ -484,12 +466,13 @@ private ResourceBundle getResourceBundleInternal(ResourceResolver optionalResolv
                         }
 
                         resourceBundle = createResourceBundle(optionalResolver, key.baseName, key.locale);
-                        // put the newly created ResourceBundle to the cache. If it replaces an existing entry unregister the existing
-                        // service registration first before re-registering the new ResourceBundle.
-                        if (resourceBundleCache.put(key, resourceBundle) != null) {
-                            unregisterResourceBundle(key);
-                        }
-                        registerResourceBundle(key, resourceBundle);
+                        resourceBundleRegistry.updateRB(key, resourceBundle);
+
+                        final Set<String> languageRoots = resourceBundle.getLanguageRootPaths();
+                        this.languageRootPaths.addAll(languageRoots);
+
+                        log.debug("Key {} - added service registration and language roots {}", key, languageRoots);
+                        log.info("Currently loaded dictionaries across all locales: {}", languageRootPaths);
```

Comment:
Should these messages stay, be removed or both be on debug level?

Reply:
They are legacy, so I leave them as is.